File: exercism/typescript/perfect-numbers/perfect-numbers.ts
export function classify(num: number): string {
  if (num <= 0) {
    throw new Error("Classification is only possible for natural numbers.")
  }

  let factors: number[] = []
  if (num === 1) {
    return "deficient";
  } else {
    for (let i = 0; i <= num / 2; i++) {
      if (num % i === 0) {
        factors.push(i)
      }
    }
  }

  let aliquotSum = factors.reduce((x, y) => x + y);

  if (aliquotSum === num) {
    return "perfect";
  } else if (aliquotSum > num) {
    return "abundant";
  } else {
    return "deficient";
  }
}

File: exercism/python/perfect-numbers/perfect_numbers.py
def classify(number: int) -> str:
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")

    aliquot_sum: int = sum(factors(number))
    if aliquot_sum == number:
        return "perfect"
    elif aliquot_sum > number:
        return "abundant"
    elif aliquot_sum < number:
        return "deficient"


def factors(number: int) -> [int]:
    """ Calculates all the factors of the given number (not including the number itself)
    :param number: int - a positive integer
    :return :[int] - list of all factors for a givet number
    """
    return [x for x in range(1, number // 2 + 1) if number % x == 0]
